File: app/src/main/java/com/johnny/gank/ui/activity/BaseActivity.java
package com.johnny.gank.ui.activity;
/*
 * Copyright (C) 2016 Johnny Shieh Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import com.johnny.gank.GankApplication;
import com.johnny.gank.di.component.AppComponent;

import android.app.Fragment;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

/**
 * description
 *
 * @author Johnny Shieh (JohnnyShieh17@gmail.com)
 * @version 1.0
 */
public class BaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    public AppComponent getAppComponent() {
        return ((GankApplication)getApplication()).getAppComponent();
    }

    public void replaceFragment(int containerViewId, Fragment fragment, String tag) {
        if(null == getFragmentManager().findFragmentByTag(tag)) {
            getFragmentManager().beginTransaction()
                .replace(containerViewId, fragment, tag)
                .commit();
        }
    }
}

File: app/src/main/kotlin/com/johnny/gank/ui/activity/BaseActivity.kt
package com.johnny.gank.ui.activity
/*
 * Copyright (C) 2016 Johnny Shieh Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import android.app.Fragment
import android.support.v7.app.AppCompatActivity

/**
 * description
 *
 * @author Johnny Shieh (JohnnyShieh17@gmail.com)
 * @version 1.0
 */
abstract class BaseActivity : AppCompatActivity() {

    fun replaceFragment(containerViewId: Int, fragment: Fragment, tag: String) {
        if (null == fragmentManager.findFragmentByTag(tag)) {
            fragmentManager.beginTransaction()
                    .replace(containerViewId, fragment, tag)
                    .commit()
        }
    }
}
